docs/workloads/azure/data/getting_started/etl_pipelines_deployment_azure.md

```diff
@@ -1,17 +1,128 @@
+
+* Azure Data Factory resources (defined in Terraform / ARM)
+    * Linked service
+    * Dataset
+    * Pipeline
+* Data ingest config files (JSON)
+* Azure DevOps CI/CD pipeline (YAML)
+* (optional) Spark jobs for data quality tests (Python)
+* Template unit tests (Python)
+* Template end-to-end tests (Python, Behave)
+
+## Create feature branch
+
+Before creating a new workload using Datastacks, open the project locally and create a new branch for the workload being created, e.g.:
+
+```bash
+git checkout -b feat/my-new-data-pipeline
+```
+
+## Prepare the Datastacks config file
+
+Datastacks requires a config file for generating a new ingest workload. This config file should be a yaml file. A sample config file is included in the [de_templates](https://github.com/amido/stacks-azure-data/tree/main/de_templates) folder.
+
+Using the sample config file as a starting point, create a new config file and populate the values relevant to your new ingest pipeline, e.g.
+
+```yaml
+# `dataset_name` parameter is used to determine names of the following ADF resources:
+# - pipeline: Ingest_<dataset_name>
+# - dataset: ds_<dataset_name>
+# - linked service: ls_<dataset_name>
+dataset_name: AzureSql_MyNewExample
+pipeline_description: "Ingest from demo Azure SQL database using ingest config file."
+data_source_type: azure_sql
+
+key_vault_linked_service_name: ls_KeyVault
+data_source_password_key_vault_secret_name: sql-password
+data_source_connection_string_variable_name: sql_connection
+
+# Azure DevOps configurations
+
+ado_variable_groups_nonprod:
+  - amido-stacks-de-pipeline-nonprod
+  - amido-stacks-infra-credentials-nonprod
+  - stacks-credentials-nonprod-kv
+
+ado_variable_groups_prod:
+  - amido-stacks-de-pipeline-prod
+  - amido-stacks-infra-credentials-prod
+  - stacks-credentials-prod-kv
+
+# Datalake containers
+
+bronze_container: raw
+silver_container: staging
+gold_container: curated
+```
+
+## Generate project artifacts using Datastacks
+
+Use the Datastacks CLI to generate the artifacts for the new workload, using the prepared config file (replacing `path_to_config_file/my_config.yaml` with the appropriate path). Note, a workload with Data Quality steps requires a data platform with a Databricks workspace:
+
+```bash
+# Initiate Datastacks using poetry:
+poetry run datastacks
+
+# Generate resources for an ingest pipeline (without Data Quality steps)
+datastacks generate ingest --config="path_to_config_file/my_config.yaml"
+
+# Generate resources for an ingest pipeline (with added Data Quality steps)
+datastacks generate ingest --config="path_to_config_file/my_config.yaml" --data-quality
+```
+
+This should add new project artifacts for the workload under `de_workloads/ingest`, based on the ingest workload templates. Review the resources that have been generated.
+
+Once reviewed, commit the resources and push the branch to the remote project repository.
+
+## Deploy new workload in non-production environment
```

Comment:
The documentation looks good, but I think that the QA user story should include not only deploying the pipelines, but also testing whether they can be successfully executed in ADF. Maybe update this? https://amido-dev.visualstudio.com/Amido-Stacks/_workitems/edit/6535
Otherwise feel free to resolve.

Reply:
Have added a note to this story.